src/prng/os-prng.lisp

```diff
@@ -0,0 +1,25 @@
+;;;; -*- mode: lisp; indent-tabs-mode: nil -*-
+;;;; os-prng.lisp -- OS-provided pseudo-random number generator
+
+(in-package :crypto)
+
+(defclass os-prng ()
+  (#+unix source))
+
+(defmethod prng-random-data (num-bytes (prng os-prng))
+  #+unix
+  (let ((seq (make-array num-bytes :element-type 'unsigned-byte)))
+    (unless (slot-boundp prng 'source)
+      (setf (slot-value prng 'source)
+            (open #P"/dev/urandom" :element-type 'unsigned-byte)))
```

Comment:
Here the ```:element-type``` being ```'unsigned-byte``` makes the function return a ```(simple-vector n)```, which causes some type issues with functions requiring a ```(simple-array (unsigned-byte 8) (n))``` (e.g. key pair generation).

Changing ```:element-type 'unsigned-byte``` to ```:element-type '(unsigned-byte 8)``` solves the problem.